### PR TITLE
Removed all the dead code.

### PR DESCRIPTION
**Pull request by daidoji**
## Rationale
Just trying to solve the issue to get familiar with the rust tools, the build process, and the warnings.  

Simply made warnings -> errors and then removed all the dead code.  This may not have been the right thing to do and if so, def let me know what I should have done :D  I didn't really ask for clarity so totally glad to redo it with a little direction.

## Testing
Ran the tests that were already present and none seemed to break after I removed all this code.

### For Reviewers
- [ ] Verify version bumped in `Cargo.toml`

### CODE REVIEW

**Comment by daidoji** [comment]
Also we verify that we manually bump the version in these PRs but should we also update the edition to the current year?

File: src/core/bexter.rs [hunk 1]
```diff
@@ -311,16 +311,4 @@ mod test {
         assert_eq!(bexter.qb2().unwrap(), qb2);
         assert_eq!(bexter.bext().unwrap(), bext);
     }
-
-    fn unhappy() {
```
Comment (jasoncolburne) [inline src/core/bexter.rs:315]: Instead, add an #[test] tag above

File: src/core/common.rs [hunk 2]
```diff
@@ -102,26 +102,6 @@ pub mod Ids {
 const REVER_STRING: &str = "(?P<ident>[A-Z]{4})(?P<major>[0-9a-f])(?P<minor>[0-9a-f])(?P<kind>[A-Z]{4})(?P<size>[0-9a-f]{6})_";
 const IDENTS: &[&str] = &[Identage::ACDC, Identage::KERI];
 const SERIALS: &[&str] = &[Serialage::JSON];
-const ILKS: &[&str] = &[
```
Comment (jasoncolburne) [inline src/core/common.rs:105]: I believe I use these externally

File: src/lib.rs [hunk 3]
```diff
@@ -1,5 +1,4 @@
-// TODO: remove before 1.0.0
-#![allow(dead_code)]
+#![deny(warnings)]
```
Comment (jasoncolburne) [inline src/lib.rs:1]: You can simply remove this I think

**Review by jasoncolburne — COMMENTED** [review]
Just a couple comments

File: src/core/common.rs (hunk 2 above)
Comment (jasoncolburne) [reply src/core/common.rs:105]: I may be wrong though, I may just use `Ilkage` actually

File: src/lib.rs (hunk 3 above)
Comment (jasoncolburne) [reply src/lib.rs:1]: Oh sorry, I see the change. Nice

File: src/core/common.rs (hunk 2 above)
Comment (daidoji) [reply src/core/common.rs:105]: Oh I wasn't aware that this library was in use yet.  Is it consumed in one of the public WebOfTrust tools or in a private project?

**Comment by daidoji** [comment]
Added back the `unhappy` function and notated it as a test.

File: src/core/common.rs (hunk 2 above)
Comment (jasoncolburne) [reply src/core/common.rs:105]: It's consumed by my company, Qui Identity. I have a KERI/ACDC implementation in Rust in a library we make use of.

Comment (qui-jason) [reply src/core/common.rs:105]: Just checked, I do only use Ilkage:

```
❯ rg ILKS .
❯ rg Ilkage .
./src/keri/verification.rs
4:use cesride::{common::Ilkage, data::Value, Matter, Sadder, Saider, Seqner, Serder};
11:        Ilkage::vcp => serder.pre(),
12:        Ilkage::iss | Ilkage::rev => serder.ked()["ri"].to_string(),

./src/keri/parsing.rs
4:    common::{Identage, Ilkage},
47:                        Ilkage::icp | Ilkage::rot | Ilkage::ixn => {
77:                        Ilkage::vcp | Ilkage::iss | Ilkage::rev => match group {
... and so forth
```

**Review by jasoncolburne — APPROVED** [review]
Looks good!

**Review by m00sey — APPROVED** [review]

**Comment by m00sey** [comment]
Closes #60

**Pull request merged**